controllers/event.js

```diff
@@ -129,59 +129,62 @@ router.put("/:id/edit", (req, res) => {
 	});
 });
 
-// Add participant and decrement limit
+// Add participant
 router.put("/:id/join", (req, res) => {
-	// Check if organiser attempts to join own event
-	Event.findOne(
-		{
-			organiser: req.user._id,
-		},
-		(err, foundEvent) => {
-			if (foundEvent)
-				res.status(403).json({
-					error: "Organiser cannot be added to participants",
-				});
-			else {
-				// Check if participant has joined event
-				Event.findOne(
-					{ participants: { $in: req.user._id } },
-					(err, foundEvent) => {
-						if (foundEvent)
-							res.status(403).json({
-								error: "Participant already joined event",
-							});
-						else {
-							// Update event by adding particpant and decrement limit
-							Event.findByIdAndUpdate(
-								req.params.id,
-								{ $push: { participants: req.user._id } },
-								{ new: true },
-								(err, updatedEvent) => {
-									if (err) res.status(500).json({ error: err });
-									else {
-										res.status(201).json(updatedEvent);
-									}
+	// Check if participant has joined event
+	Event.findOne({ participants: { $in: req.user._id } }, (err, foundEvent) => {
```

Comment:
Hey Wenjie,

Similar as initial code. If you use findOne and only check participants without checking event id, the result is not related to that particular event, it becomes a false error.

Reply:
Hey Jess,

I've removed the participant query altogether. 
Have also updated user profile to retrieve the event object instead of restricting to  id and event title.